### PR TITLE
Roomba Cleanbots

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Adds a New Roomba Sprite to Clean Bots

They are animated when cleaning(Every Direction). I do not know how to make a gif of them in action.
They are also directional!

## Why It's Good For The Game
Bucket Sprite looks dumb. Much better sprite.

## TODO

- [x] Keep Bucket Sprites in addition to Roomba Sprites.
- [x] Make a Roomba Subtype
- [x] Add a Roomba Frame
- [x] Make a recipe for Roomba using Roomba frame.

## The Sprites

<details>

<summary>Sprites</summary>

RoombaOff

RoombaOn

</details>

## Changelog
:cl: PestoVerde322, DatBoiTim
add: Roomba Subtype of Cleanbot and associated recipe to make them
add: Roomba frame for constructing Roombas
imageadd: New Roomba Sprite for Cleanbots, courtesy of Pestoverde322
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
But... They're made *with* buckets.

**Comment by DatBoiTim** [comment]
> But... They're made _with_ buckets.

Beepskies are made from helmets. They do not look like helmets.

And what is a bucket if not a very tall roomba?

**Comment by Crossedfall** [comment]
> Beepskies are made from helmets. They do not look like helmets.

My man, it's just an upside-down helmet. It even has the visor. 

image

**Review by Crossedfall — DISMISSED** [review]
Make this an alternative build recipe where you need to make/print a frame.

**Comment by DatBoiTim** [comment]
"Roombas Now Coexist with Buckets"

A Statement that only a game like SS13 would produce

**Comment by DatBoiTim** [comment]
Gotta figure out why design for frame isn't showing.

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm [hunk 1]
```diff
@@ -276,11 +276,108 @@
...
+		playsound(src, 'sound/effects/spray2.ogg', 50, 1, -6)
+		A.acid_act(75, 10)
+	else if(istype(A, /mob/living/simple_animal/cockroach) || istype(A, /mob/living/simple_animal/mouse))
+		var/mob/living/simple_animal/M = target
+		if(!M.stat)
+			visible_message("<span class='danger'>[src] smashes [target] with its mop!</span>")
+			M.death()
+		target = null
+
+	else if(emagged == 2) //Emag functions
```
Comment (Archanial) [inline code/modules/mob/living/simple_animal/bot/cleanbot.dm:342]: please just use the emagged flag

**Review by Archanial — APPROVED** [review]
Next please add roomba with knife

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm (hunk 1 above)
Comment (DatBoiTim) [reply code/modules/mob/living/simple_animal/bot/cleanbot.dm:342]: These are mobs. They don't have a field for this.

**Comment by DatBoiTim** [comment]
> Next please add roomba with knife

Is planned, already have a branch for this staged.

**Review by Archanial — APPROVED** [review]

**Pull request merged**